src/interactive-window/editor-integration/cellHashProviderFactory.ts

```diff
@@ -7,8 +7,9 @@ import { IConfigurationService } from '../../platform/common/types';
 import { IServiceContainer } from '../../platform/ioc/types';
```

Comment:
@rchiodo @IanMatthewHuff Here's a rough implementation of what was discussed.
Still some work to be done, please review and let me know.

TODO:
* We need to ensure the cell hashes are generated fo IW before we execute (in the `createExecutionPromise` of `InteractiveWindow.node.ts` file).
* Need to get all of the hashes and send them in this same plaec.

@rchiodo I don't think we need the `willExecute` hook, as the cell hashes are very specific to IW and doesn't apply to notebooks.

Anyways, this is still a WIP. But hopefully the architecture is as discussed and looks like.